doc/adrs/115_split_sirius_web_frontend.adoc

```diff
@@ -0,0 +1,70 @@
+= ADR-115 - Split Sirius Web frontend
```

Comment:
I agree on the general principle of course, I would simply recommend a couple of small changes:

I would name the newly created module `sirius-web-application` because `sirius-web-core` kinda implies that it's a small central thing while this module will end up being massive AND it will be split in the future. The way I see it, people who want to reuse pretty much everything but customize some minor details will just depend on `sirius-web-application` which in turn will depend in the future on something like `sirius-web-router`, `sirius-web-projects`, `sirius-web-widgets`, `sirius-web-core`.

This is similar to what we have on the backend with the `sirius-components-starter` which brings pretty much everything while `sirius-components-core` is a very small core module.

Reply:
It would also nice to indicate that we will more quickly on this part. Since nothing in Sirius Web was really an API to be extended but instead an example to be forked and now we will start to introduce Sirius Web APIs to extend this application. We will not get it right the first time and thus break APIs from `sirius-web-application` repeatedly in the near future.

For example, we will have to think carefully about what is API in this new frontend package and how it is used by the existing `sirius-web` package. For example, you are mentioning here that the footer could be extracted. While I agree on principle, in reality this would require quite some changes. That would be the same thing for the navigation bar which is referenced deeply by all the pages etc.

Moving forward this way would result in tons of changes in one PR. For that, I think that you should start with one PR introducing `sirius-web-application` and moving everything inside and only exporting in its `index.ts` the views:

```
export { EditProjectView } from '../views/edit-project/EditProjectView';
export { NewProjectView } from '../views/new-project/NewProjectView';
export { ProjectSettingsView } from '../views/project-settings/ProjectSettingsView';
export { ProjectsView } from '../views/projects/ProjectsView';
export { UploadProjectView } from '../views/upload-project/UploadProjectView';
```

and nothing more. Then we will have to figure out "what do we want to configure" and "how to make it configurable. And this would be a very bad first version since consumers would have to copy and reconnect everything in the current `index.ts` of `sirius-web` with, for example:

- The ApolloProvider
- The BrowserRouter
- The ThemeProvider
- The ServerContext
- The ToastProvider
- The RepresentationContext.Provider
- The PropertySectionContext.Provider
- The main style

For that, I think we will need to expose something similar to the `Workbench` component of Sirius Components with something like this:

```
<SiriusWebApplication>
  <Views navigationBar={<CustomNavigationBar />} footer={<CustomFooter />}>
    <MyCustomView />
  </Views>
  <FormWidgets>
    <MyCustomWidget />
  </FormWidgets>
</SiriusWebApplication>
```